### ISSUE TITLE
Disable auto acks

### ISSUE THREAD

**Issue by tekjar** [AtherEnergy/rumqtt#101]
#94 

Sometimes users might want to send acks **after** handling incoming publishes. Add compile time feature to disable auto acks and provide method to send ack

**Comment by stanmueller** [comment]
I have a suggestion how to fix the "auto acks" problem.
I suggest to extent the enum Notification:: Publish(Publish) with information needed for confirming a received message. => enum Notification:: Publish(Publish,**AckInfo**) and not, as now, confirming incoming message immediately.
After the Notification:: Publish(Publish,**AckInfo**) was processed in the user code, Rust calls drop() method for the AckInfo struct. Within this method should the acknowledgement be sent to the Brocker. For this AckInfo must implement the Dorp Traid.
This is only the rough approach to solve this problem. I'm too new in rust to fix the problem myself.

**Comment by tekjar** [comment]
@stanmueller Yeah that sounds plausible. But I should say that disabling auto-ack isn't a priority for me now. Will come back to this after next release. But you are welcome to give it a shot now :). I would be interested in how it looks
